solr/parser.go

```diff
@@ -269,3 +269,51 @@ func (parser *MoreLikeThisParser) Parse(resp_ *[]byte) (*SolrMltResult, error) {
 	}
 	return sr, nil
 }
+
+type AnyResultParser interface {
+	Parse(*[]byte) (*SolrAnyResult, error)
+}
+
+type AnyParser struct {
+}
+
+func (parser *AnyParser) Parse(resp_ *[]byte) (*SolrAnyResult, error) {
+	jsonbuf, err := bytes2json(resp_)
+	sr := &SolrAnyResult{}
+	if err != nil {
+		return sr, nil
+	}
+	var resp = new(SolrResponse)
+	resp.Response = jsonbuf
+	resp.Status = int(jsonbuf["responseHeader"].(map[string]interface{})["status"].(float64))
+
+	sr.Results = new(Collection)
+	sr.Status = resp.Status
+	sr.Payload = make(map[string]interface{})
+
+	if responseHeader, ok := resp.Response["responseHeader"].(map[string]interface{}); ok {
+		sr.ResponseHeader = responseHeader
+	}
+
+	if resp.Status != 0 {
+		if err, ok := resp.Response["error"].(map[string]interface{}); ok {
+			sr.Error = err
+		}
+		return sr, err
+	}
+
+	for k, r := range resp.Response {
+		switch k {
+		case "response":
```

Comment:
Do we think that we still have some kind of response that is corresponding to collection. My thought is that we should skip this and put everything i Payload. We do very generic stuff here and let application go for specific parsing

Reply:
Look like suggester didn't have response?

Reply:
> Do we think that we still have some kind of response that is corresponding to collection. My thought is that we should skip this and put everything i Payload. We do very generic stuff here and let application go for specific parsing

Since it is still part of Solr responds then it be parsed using the same methods in the library. Only if it is new response then it is left up to the developer to parse it.

> Look like suggester didn't have response?

No the suggest response would look something like this: 
```
{
   "responseHeader":{
      "status":0,
      "QTime":2
   },
   "suggest":{
      "suggester":{
         "keyword":{
            "numFound":0,
            "suggestions":[

            ]
         }
      }
   }
}
```

Reply:
And you're sure that response always contains documents response, ie

`  "response":{"numFound":1,"start":0,"maxScore":2.3025851,"docs":[
      {}]`

Reply:
Mostly yes unless Solr change their response scheme